archive/tar.go

```diff
@@ -602,6 +602,11 @@ func (cw *ChangeWriter) HandleChange(k fs.ChangeKind, p string, f os.FileInfo, e
 		hdr.AccessTime = time.Time{}
 		hdr.ChangeTime = time.Time{}
 
+		// uname and gname strings are from the host filesystem and meaningless for containers
+		// (https://github.com/moby/buildkit/issues/3688)
+		hdr.Uname = ""
+		hdr.Gname = ""
```

Comment:
Should we disable the lookup altogether? In moby/moby we previously had a fork of `pkg/archive` to disable lookup of usernames (related to [https://github.com/advisories/GHSA-v2cv-wwxq-qq97](https://github.com/advisories/GHSA-v2cv-wwxq-qq97)), but in https://github.com/moby/moby/commit/e9bbc41dd146d692e28660a392b068c9c112f2ad, we implemented a new approach to disable it (while using Go's stdlib)

/cc @corhere

Reply:
ISTR the lookup not just looks up in `/etc/passwd`, but (depending on the host) also looks up in other ways.

Not sure if it applies to this as well, but I recall cases on Windows where this could even result in connecting to Active Directory (something which has been known to cause 10 seconds or more delays in some setups).

Reply:
Updated